src/site/content/en/blog/slow-roads/index.md

```diff
@@ -0,0 +1,237 @@
+capable and convenient tools and libraries. It's now easier than ever for front-end developers to incorporate 3D UX into their
+work, even without prior experience in computer graphics. [Three.js](https://threejs.org/), the leading WebGL library, serves
+as the foundation for many expansions, including react-three-fiber which brings 3D components into the React framework. There
+are now also comprehensive web-based game editors such as [Babylon.js](https://www.babylonjs.com/) or [PlayCanvas](https://playcanvas.com/)
+which offer a familiar interface and integrated toolchains.
+
+Despite the remarkable utility of these libraries, however, ambitious projects are eventually bound by technical limitations. Skeptics
+to the idea of browser-based gaming might highlight that JavaScript is single-threaded and resource-constrained. But navigating
+these limitations unlocks the hidden value: no other platform offers the same instant accessibility and mass compatibility
+enabled by the browser. Users on any browser-capable system can begin playing in one click, with no need to install applications
+and no need to sign in to services. Not to mention that developers enjoy the elegant convenience of having robust front-end frameworks
+available for building UI, or handling networking for multiplayer modes. These values, in my opinion, are what make the browser such an
+excellent platform for both players and developers alike—and, as demonstrated by Slow Roads, the technical limitations might often be
+reducible to a design problem.
+
+## Achieving smooth performance in Slow Roads
+
+Since the core elements of Slow Roads involve high-speed motion and expensive scenery generation, the need for smooth performance
+underlined my every design decision. My main strategy was to start with a pared-down gameplay design that allowed for contextual
+short-cuts to be taken within the engine's architecture. On the downside this means trading off some nice-to-have features in the
+pursuit of minimalism, but results in a bespoke, hyper-optimized system that plays nicely across different browsers and devices.
+
+Here follows a breakdown of the key components that keep Slow Roads lean.
+
+### Shaping the environment engine around the gameplay
+
+As the key component of the game, the environment generation engine is unavoidably expensive, justifiably taking the greatest
+proportion of the budgets for memory and compute. The trick used here is in scheduling and distributing the heavy computation over
+a period of time, so as not to interrupt the framerate with performance spikes.
+
+{% Aside %}
+At its core, a game engine is an infinite `while` loop which reads user input, updates the world state, and then draws the
+image at 60 frames per second. As JavaScript is single threaded, any computation happening within these times needs to complete
+within the given frame time, or the user will experience visible stutters.
```

Comment:
```suggestion
At its core, a game engine is an infinite `while` loop which reads user input, updates the world state, and then renders the image for the frame. This loop runs at the refresh rate of the monitor (typically 60 Hz), so any computation happening within it needs to complete within the given frame time to avoid causing visible stutters.
```